### PR TITLE
Legend shows arrow from annotate #8236

### PR DESCRIPTION
**Pull request by Wojciech-Lange**
## PR Summary

Answer for #8236 issue. Legend automatically handles an arrow from annotate. In fact the code is a limited variant of #10688 . FancyArrowPatch and HandlerAnnotation added. There might be some relicts left from the wider solution (#10688).

Example code:

```python
import matplotlib.pylab as plt
from matplotlib.patches import FancyArrowPatch
from matplotlib.legend_handler import HandlerAnnotation

fig, ax = plt.subplots(1)
ax.plot([0, 1], [0, 0], label='line1')
ax.plot([0, 1], [1, 1], label='line2')
ax.annotate("",
            xy=(0.3,1.0),
            xytext=(0.3,0.0),
            arrowprops={'arrowstyle':'<->', 'color':'C7' },
            label='distance')
ax.legend()

plt.show()
```
example

## PR Checklist

- [x] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant 
- [x] Example in examples folder
- [x] (probably) Documentation is sphinx and numpydoc compliant

### CODE REVIEW

File: lib/matplotlib/legend_handler.py [hunk 1]
```diff
@@ -723,3 +724,77 @@ def create_artists(self, legend, orig_handle,
...
+        width_ratios=[1, 4],
+        **kwargs
+    ):
+
+        self._pad = pad
+        self._width_ratios = width_ratios
+
+        HandlerBase.__init__(self, **kwargs)
+
+    def create_artists(
```
Comment (dstansby) [inline lib/matplotlib/legend_handler.py]: Could the call aruguments be on multiple lines, but with a maximum line length of 79 characters? (if that makes sense; like
```python
def func(arg1, arg2, arg3, arg4,
         arg5)
```

File: lib/matplotlib/legend_handler.py [hunk 2]
```diff
@@ -723,3 +724,77 @@ def create_artists(self, legend, orig_handle,
...
+    ):
+
+        if orig_handle.arrow_patch is not None:
+
+            # Arrow without text
+
+            handler = HandlerFancyArrowPatch()
+            handle = orig_handle.arrow_patch
+
+        return handler.create_artists(
```
Comment (dstansby) [inline lib/matplotlib/legend_handler.py]: Same as above regarding the function arguments here.

File: lib/matplotlib/legend.py [hunk 3]
```diff
@@ -1219,12 +1224,12 @@ def _get_legend_handles(axs, legend_handler_map=None):
...
     for ax in axs:
         handles_original += (ax.lines + ax.patches +
-                             ax.collections + ax.containers)
+                             ax.collections + ax.containers+ax.texts)
         # support parasite axes:
         if hasattr(ax, 'parasites'):
             for axx in ax.parasites:
                 handles_original += (axx.lines + axx.patches +
-                                     axx.collections + axx.containers)
+                                     axx.collections + axx.containers+ax.texts)
```
Comment (timhoffm) [inline lib/matplotlib/legend.py]: shouldn't that be `axx.texts`?

**Comment by jklymak** [comment]
@Haloxxx  this looks quite close, but it'd help if you addressed the comments above.

**Comment by Haloxxx** [comment]
> @Haloxxx this looks quite close, but it'd help if you addressed the comments above.

After these few months my request doesn't pass through checks. I've tried to apply all the suggestions, but I'm not sure if I can handle this by myself.

**Comment by QuLogic** [comment]
This looks mostly done, though could use some tests and needs flake8 fixes.

We may take this up at some point if you don't reply soon.

**Comment by tacaswell** [comment]
You may also need to rebase this on current master to pick up the fixes / changes to the doc build.

**Comment by dstansby** [comment]
I'm going to close this in favour of https://github.com/matplotlib/matplotlib/pull/23160, which is an updated version of this PR.

**Pull request closed without merging**